kernel/src/table_changes/scan_file.rs

```diff
@@ -0,0 +1,447 @@
+//! This module handles [`CDFScanFile`]s for [`TableChangeScan`]. A [`CDFScanFile`] consists of all the
+//! metadata required to generate a change data feed. [`CDFScanFile`] can be constructed using
+//! [`CDFScanFileVisitor`]. The visitor reads from engine data with the schema [`cdf_scan_row_schema`].
+//! You can convert engine data to this schema using the [`get_cdf_scan_row_expression`].
+use itertools::Itertools;
+use std::collections::HashMap;
+use std::sync::{Arc, LazyLock};
+
+use super::log_replay::TableChangesScanData;
+use crate::actions::visitors::visit_deletion_vector_at;
+use crate::engine_data::{GetData, TypedGetData};
+use crate::expressions::{column_expr, Expression};
+use crate::scan::state::DvInfo;
+use crate::schema::{
+    ColumnName, ColumnNamesAndTypes, DataType, MapType, SchemaRef, StructField, StructType,
+};
+use crate::utils::require;
+use crate::{DeltaResult, EngineData, Error, RowVisitor};
+
+/// Transforms an iterator of TableChangesScanData into an iterator  of
+/// `(CDFScanFile, Option<Arc<HashMap<String, DvInfo>>>)`. The `CDFScanFile` is constructed by
+/// visiting the engine data.
+#[allow(unused)]
+pub(crate) fn scan_data_to_scan_file(
+    scan_data: impl Iterator<Item = DeltaResult<TableChangesScanData>>,
+) -> impl Iterator<Item = DeltaResult<(CDFScanFile, Option<Arc<HashMap<String, DvInfo>>>)>> {
```

Comment:
I was considering introducing two types:
```
#[allow(unused)]
pub(crate) struct UnresolvedCDFScanFile {
    scan_file: CDFScanFile,
    remove_dvs: Arc<HashMap<String, DvInfo>>,
}
#[allow(unused)]
pub(crate) struct ResolvedCDFScanFile {
    scan_file: CDFScanFile,
    selection_vector: Vec<bool>,
}
```

Because the step after this will be to resolve deletion vectors. Is this sensible naming?